db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -1,38 +1,221 @@
+            getExhibitions().subscribe({
+                if (it is ProgressDataState.Done<*>) {
+                    currentDownloads++
+                    observer.onNext(currentDownloads)
+                }
+            }, {
+                observer.onError(it)
+            }, {
+                if (currentDownloads == maxDownloads) {
+                    observer.onComplete()
+                }
+            })
+            getEvents().subscribe({
+                if (it is ProgressDataState.Done<*>) {
+                    currentDownloads++
+                    observer.onNext(currentDownloads)
+                }
+            }, {
+                observer.onError(it)
+            }, {
+                if (currentDownloads == maxDownloads) {
+                    observer.onComplete()
+                }
+            })
+        }
+
+    }
+
+    private fun getExhibitions(): Observable<ProgressDataState> {
+        return serviceProvider.getExhibitions()
                 .flatMap {
-                    val dtf = DateTimeFormatterBuilder()
-                            //Thu, 12 Jul 2018 16:56:12 GMT
-                            .appendPattern("E, dd MMM yyyy HH:mm:ss O")
-                            .toFormatter()
-                    if (it.containsKey("last-modified")) {
-                        val lastModified = it["last-modified"]?.get(0)!!
-                        val newLocalDateTime = LocalDateTime.parse(lastModified, dtf)
-                        if(newLocalDateTime.isAfter(/*TODO: get from storage*/ LocalDateTime.parse("Thu, 12 Jul 2018 16:56:12 GMT", dtf))) {
-                            return@flatMap serviceProvider.getBlob()
-                        } else {
-                            return@flatMap Observable.just(AppDataState.Empty)
+                    when (it) {
+                        is ProgressDataState.Done<*> -> {
+                            val result = it.result as ArticResult<*>
+                            if (result.data.isNotEmpty() && result.data[0] is ArticExhibition) {
+                                @Suppress("UNCHECKED_CAST")
+                                appDatabase.exhibitionDao.updateExhibitions((result as ArticResult<ArticExhibition>).data)
+
+                            }
                         }
-                    } else {
-                        serviceProvider.getBlob()
-                                .doOnNext { 
-                                    //TODO: save date time here
-                                }
                     }
 
-                }.doOnNext {
-                    if (it is AppDataState.Done) {
-                        //Save to db
+                    it.asObservable()
+                }
+    }
+
+
+    private fun getEvents(): Observable<ProgressDataState> {
+        return serviceProvider.getEvents()
+                .flatMap {
+                    when (it) {
+                        is ProgressDataState.Done<*> -> {
```

Comment:
same as above. it would cleanup duplicated code. maybe in a `retrofit-rx-extensions` module or something.